src/views/Nft/market/hooks/useGetCollectionDistribution.tsx

```diff
@@ -29,24 +33,56 @@ const useGetCollectionDistribution = (collectionAddress: string) => {
 
 interface StatePB {
   isFetching: boolean
-  data: ApiCollectionDistributionPB['data']
+  total: number
+  data: Record<string, ApiSingleTokenData & { tokenCount: number }>
 }
 
 export const useGetCollectionDistributionPB = () => {
-  const [state, setState] = useState<StatePB>({ isFetching: false, data: null })
+  const [state, setState] = useState<StatePB>({ isFetching: false, total: 0, data: null })
 
   useEffect(() => {
     const fetchTokens = async () => {
       setState((prevState) => ({ ...prevState, isFetching: true }))
-      const apiResponse = await getCollectionDistributionApi<ApiCollectionDistributionPB>(pancakeBunniesAddress)
-      setState({
-        isFetching: false,
-        data: apiResponse.data,
-      })
+      const apiResponse = await getNftsFromCollectionApi(pancakeBunniesAddress)
+      const tokenIds = Object.keys(apiResponse.attributesDistribution)
+      const totalCountCall = {
+        address: getPancakeRabbitsAddress(),
+        name: 'totalSupply',
+      }
+      const bunnyCountCalls = tokenIds.map((tokenId) => ({
+        address: getPancakeRabbitsAddress(),
+        name: 'bunnyCount',
+        params: [tokenId],
+      }))
+      const bunnyContractCalls = [totalCountCall].concat(bunnyCountCalls)
+      try {
+        const response = await multicallv2(pancakeRabbitsAbi, bunnyContractCalls)
+        const tokenListResponse = response.slice(1).reduce((obj, tokenCount, index) => {
+          return {
+            ...obj,
+            [tokenIds[index]]: { ...apiResponse.data[index], tokenCount: tokenCount[0].toNumber() },
+          }
+        }, {})
+        setState({
+          isFetching: false,
+          total: response[0][0].toNumber(),
+          data: tokenListResponse,
+        })
+      } catch (error) {
+        // Use graph data if multicall fails
```

Comment:
Let's wrap it another try/catch, `getNftsFromCollectionApi` may also fail.

Reply:
maybe document the purpose of graph + multicall 🙂

Reply:
> maybe document the purpose of graph + multicall 🙂

Done